File: model/src/main/java/com/itransition/fanfictionbackend/model/lookup/ReactionLookUp.java
package com.itransition.fanfictionbackend.model.lookup;

import com.itransition.fanfictionbackend.model.Reaction;
import lombok.Data;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.Table;

@Data
@Entity
@Table(name = "reaction_lut")
public class ReactionLookUp implements Serializable {

    @Id
    private String name;

    @OneToMany(mappedBy = "value", orphanRemoval = true)
    private List<Reaction> reaction = new ArrayList<>();
}

File: src/main/kotlin/com/itransition/fanfictionbackend/model/lookup/ReactionLookUp.kt
package com.itransition.fanfictionbackend.model.lookup

import com.itransition.fanfictionbackend.model.Reaction
import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.OneToMany
import javax.persistence.Table

@Entity
@Table(name = "reaction_lut")
class ReactionLookUp {

    @Id
    var name: String? = null

    @OneToMany(mappedBy = "value", orphanRemoval = true)
    var reaction: List<Reaction> = listOf()
}
